src/pip/_internal/index/package_finder.py

```diff
@@ -544,8 +552,7 @@ def sort_best_candidate(
         """
         if not candidates:
             return None
-        best_candidate = max(candidates, key=self._sort_key)
-        return best_candidate
+        return max(candidates, key=self._sort_key)
```

Comment:
IIRC, this was split for clarity on what is being returned. Let's keep that; CPython should optimise this away anyway.

Reply:
If you look at the profile all of the time is going into `_sort_key` which is reached from the sorted of `get_applicable_candidates` and the max here.  It seems redundant to have both (both in [`compute_best_candidate`](https://github.com/pypa/pip/blob/d08f968eaffe26a866fc388ffa56c9087186f03a/src/pip/_internal/index/package_finder.py#L558-L560)).  Your comment made me think of that because it's what I was wondering when I touched this.